x-pack/solutions/search/plugins/enterprise_search/public/applications/shared/kibana/kibana_logic.ts

```diff
@@ -159,10 +165,20 @@ export const KibanaLogic = kea<MakeLogicType<KibanaValues>>({
     ],
   }),
   selectors: ({ selectors }) => ({
+    isAgentlessEnabled: [
+      () => [selectors.cloud, selectors.fleet],
+      (cloud?: CloudSetup & CloudStart, fleet?: FleetStart) =>
+        (cloud?.isCloudEnabled || cloud?.isServerlessEnabled) &&
+        fleet?.config?.agentless?.enabled === true,
+    ],
     isCloud: [
       () => [selectors.cloud],
       (cloud?: CloudSetup & CloudStart) => Boolean(cloud?.isCloudEnabled),
     ],
+    isServerless: [
+      () => [selectors.cloud],
+      (cloud?: CloudSetup & CloudStart) => Boolean(cloud?.isServerlessEnabled),
```

Comment:
I think you want this:
```suggestion
      (cloud?: CloudSetup | CloudStart) => Boolean(cloud?.isServerlessEnabled),
```

Reply:
Would it be the same for `isCloud` @TattdCodeMonkey ?
`(cloud?: CloudSetup | CloudStart) => Boolean(cloud?.isCloudEnabled)`